src/Symfony/Bundle/WebProfilerBundle/EventListener/WebDebugToolbarListener.php

```diff
@@ -99,13 +99,19 @@ public function onKernelResponse(ResponseEvent $event): void
             return;
         }
 
+        // do not capture redirects done in a non-conventional way (http-equiv="refresh" tag,...).
+        $location = $response->headers->get('Location');
+        if (!$location) {
```

Comment:
Doing that outside the block handling redirection does not make sense. This removes the toolbar for all pages (this method is not just about intercepting redirections)